src/Biryuchkov/BiryuchkovTask3/coub.cpp

```diff
@@ -0,0 +1,151 @@
+#include "coub.h"
+
+Coub::Coub()
+{
+    vert = coubVertex();
+}
+
+std::vector<GLfloat> Coub::side()
+{
+    std::vector<GLfloat> vec ={};
+    for (GLfloat x = -1.0f ; x <= 1.01f-step ;x += step )
+    {
+        for (GLfloat y = -1.0f ; y <=1.01f-step ;y += step )
+        {
+            vec.push_back(x     );vec.push_back(y     );vec.push_back(1.0f);
+            vec.push_back(x+step);vec.push_back(y     );vec.push_back(1.0f);
+            vec.push_back(x     );vec.push_back(y+step);vec.push_back(1.0f);
+            vec.push_back(x     );vec.push_back(y+step);vec.push_back(1.0f);
+            vec.push_back(x+step);vec.push_back(y     );vec.push_back(1.0f);
+            vec.push_back(x+step);vec.push_back(y+step);vec.push_back(1.0f);
+        }
+    }
+    return vec;
+}
+std::vector<GLfloat> Coub::slideSide(types type)
+{
+    auto vec = side();
+    int slide = 0;
+    if (type==Z)
+    {
+        slide = 0;
+    }
+    else if(type ==Y)
+    {
+        slide = 2;
+    }
+    else if(type == X)
+    {
+        slide = 1;
+    }
+    for(size_t i = 0 ;i< vec.size()/3 ; i++)
+    {
+        std::vector<GLfloat> v = {vec[i*3],vec[i*3+1],vec[i*3+2]};
+        std::rotate(v.rbegin(), v.rbegin() + slide, v.rend());
+        for (int k = 0 ; k< 3 ; k ++)
+        {
+            vec[i*3+k] = v[k];
+        }
+
+    }
+    return vec;
+}
+std::vector<GLfloat> Coub::coubVertex()
+{
+    norm.clear();
+    std::vector<GLfloat>result = {};
+    std::vector<GLfloat> v ={};
+    std::vector<GLfloat> n ={};
+
+    v = slideSide(X);
+    result.insert(result.end() , v.begin() , v.end());
+    normals(X , v.size()/3 , false);
+
+    v = slideSide(Y);
+    result.insert(result.end() , v.begin() , v.end());
+    normals(Y , v.size()/3 , false);
+
+    v = slideSide(Z);
+    result.insert(result.end() , v.begin() , v.end());
+    normals(Z , v.size()/3 , false);
+
+    std::vector<GLfloat> reverse ={};
+
+    v = slideSide(X);
+    reverse.insert(reverse.end() , v.begin() , v.end());
+    normals(X , v.size()/3 , true);
+
+    v = slideSide(Y);
+    reverse.insert(reverse.end() , v.begin() , v.end());
+    normals(Y , v.size()/3 , true);
+
+    v = slideSide(Z);
+    reverse.insert(reverse.end() , v.begin() , v.end());
+    normals(Z , v.size()/3 , true);
+
+    for (auto it = reverse.begin(); it!= reverse.end();++it)
+    {
+        *it *=-1;
+    }
+    result.insert(result.end() , reverse.begin() , reverse.end());
+    return result;
+}
+
+void Coub::normals(types type ,int size , bool inv)

```

Comment:
```suggestion
void Coub::normals(const types type , const int size , const bool inv)

```

и в остальных функциях задания тоже нужно поправить.